src/cloud-cli/meltano/cloud/client/__init__.py

```diff
@@ -0,0 +1,164 @@
+
+if t.TYPE_CHECKING:
+    import types
+
+__all__ = ["MeltanoCloudClient"]
+
+logger = get_logger()
+
+
+class MeltanoCloudError(Exception):
+    """Base error for the Meltano Cloud API."""
+
+    def __init__(self, response: ClientResponse) -> None:
+        """Initialize the error.
+
+        Args:
+            response: The response that caused the error.
+        """
+        self.response = response
+        super().__init__(response.reason)
+
+
+class MeltanoCloudClient:
+    """Client for the Meltano Cloud API.
+
+    Attributes:
+        api_url: The API URL.
+        api_key: The API key.
+        runner_secret: The runner secret.
+        _session: The client session.
+    """
+
+    def __init__(self, api_url: str, api_key: str, runner_secret: str) -> None:
+        """Initialize the client.
+
+        Args:
+            api_url: The Cloud API URL.
+            api_key: The Cloud API key.
+            runner_secret: The runner secret.
+        """
+        self.api_url = api_url
+        self.api_key = api_key
+        self.runner_secret = runner_secret
+        self._session: ClientSession | None = None
+
+    async def __aenter__(self) -> MeltanoCloudClient:
+        """Enter the client context.
+
+        Returns:
+            The client.
+        """
+        self._session = ClientSession()
+        return self
+
+    async def __aexit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: types.TracebackType | None,
+    ) -> None:
+        """Exit the client context.
+
+        Args:
+            exc_type: The exception type.
+            exc_value: The exception value.
+            traceback: The exception traceback.
+        """
+        await self.close()
+
+    async def close(self) -> None:
+        """Close the client."""
+        if self._session:
+            await self._session.close()
+            self._session = None
+
+    @property
+    def session(self) -> ClientSession:
+        """Get the client session.
+
+        Returns:
+            The client session.
+
+        Raises:
+            RuntimeError: If the client session is not open.
+        """
+        if not self._session or self._session.closed:
+            raise RuntimeError("Client session is not open")
+        return self._session
+
+    async def _request(self, method: str, path: str, **kwargs) -> dict:
+        """Make a request to the Meltano Cloud API.
+
+        Args:
+            method: The HTTP method.
+            path: The API path.
+            **kwargs: Additional keyword arguments to pass to the request.
+
+        Returns:
+            The response JSON.
+
+        Raises:
+            MeltanoCloudError: If the response status is not OK.
+        """
+        url = f"{self.api_url}{path}"
+        headers = {
+            "Content-Type": "application/json",
+            "User-Agent": f"Meltano Cloud CLI/v{version('meltano')}",
```

Comment:
Caching the result of `version('meltano')` is probably worth doing so that we aren't reading it from disk for each request.

Reply:
Maybe, although there's currently only 1 request per invocation, so I'm not sure we'd be winning a lot from caching. Or I'm probably missing something 😅

Reply:
Nope. That's my bad.